### PR TITLE
Use internally \OCP\ILogger instead of \OC\Log

### PR DESCRIPTION
**Pull request by MorrisJobke**
- this is the preparation for some upcoming logger related changes
- also fixes an issue in the public interface where we request
  an internal class as parameter

cc @nickvergessen @DeepDiver1975 @PVince81 @rullzer @Xenopathic

@DeepDiver1975 please choose from one of the following milestones ;) 8.1 or 8.2

### CODE REVIEW

File: lib/private/backgroundjob/timedjob.php [hunk 1]
```diff
@@ -21,6 +21,7 @@
  */
 
 namespace OC\BackgroundJob;
+use OCP\ILogger;
 
 /**
  * Class QueuedJob
```
Comment (nickvergessen) [inline lib/private/backgroundjob/timedjob.php]: mind fixing that aswell?

**Comment by Xenopathic** [comment]
:+1: We have loads of references to OC_Log around the place though...

File: lib/private/updater.php [hunk 2]
```diff
@@ -67,9 +68,9 @@ class Updater extends BasicEmitter {
 	/**
 	 * @param HTTPHelper $httpHelper
 	 * @param IConfig $config
-	 * @param \OC\Log $log
+	 * @param ILogger $log
 	 */
-	public function __construct(HTTPHelper $httpHelper, IConfig $config, $log = null) {
+	public function __construct(HTTPHelper $httpHelper, IConfig $config, ILogger $log = null) {
```
Comment (DeepDiver1975) [inline lib/private/updater.php]: FYI: this will/can break the updater app - @VicDeo

Comment (nickvergessen) [inline lib/private/updater.php]: `OC\Log` implements `OCP\ILogger` even on stable8. is the type hinting really a problem?

Comment (nickvergessen) [inline lib/private/updater.php]: https://github.com/owncloud/updater/blob/master/lib/app.php#L41 looks like a non-issue to me, but lets wait for @VicDeo

**Comment by DeepDiver1975** [comment]
I'm fine with this change for 8.1 - especially because of #15965 :+1:

**Comment by DeepDiver1975** [comment]
Oooops - no Jenkins on this :imp:

**Pull request merged**